src/game-logic/input-handler.ts

```diff
@@ -15,6 +16,54 @@ export enum PlayerInput {
 
 export type PlayerAction = { playerPort: number, action: PlayerInput }
 
+export const handleCharacterSelection = (currentState: CharacterSelectionState, keysPressed: KeyStatus[]): CharacterSelectionState => {
+  const nextState: CharacterSelectionState = currentState
```

Comment:
In theory it would be preferable to make a shallow copy by using the spread operator on the state (currently changes to nextState also affect currentState). However, it's fine for now since this is the only place affecting CharacterSelectionState. 🙂